File: src/main/java/com/systemservices/kata/repository/InMemoryEmployeeRepository.java
package com.systemservices.kata.repository;

import com.systemservices.kata.model.Employee;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import org.springframework.stereotype.Repository;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicLong;

@Repository
public class InMemoryEmployeeRepository implements EmployeeRepository {

  private final Map<Long, Employee> store = new ConcurrentHashMap<>();
  private final AtomicLong seq = new AtomicLong(1);

  @Override
  public List<Employee> findAll() {
    return new ArrayList<>(store.values());
  }

  @Override
  public Optional<Employee> findById(Long id) {
    return Optional.ofNullable(store.get(id));
  }

  @Override
  public Employee save(Employee e) {
    if (e.getId() == null) {
      e.setId(seq.getAndIncrement());
    }
    store.put(e.getId(), e);
    return e;
  }

  @Override
  public void deleteById(Long id) {
    store.remove(id);
  }

  @Override
  public boolean existsById(Long id) {
    return store.containsKey(id);
  }
}

File: openspec-ts/src/repository/in-memory-employee.repository.ts
import { Injectable } from '@nestjs/common';
import { Employee } from '../model/employee';
import { EmployeeRepository } from './employee-repository.interface';

@Injectable()
export class InMemoryEmployeeRepository implements EmployeeRepository {
  private readonly store = new Map<number, Employee>();
  private sequence = 1;

  async findAll(): Promise<Employee[]> {
    return Array.from(this.store.values());
  }

  async findById(id: number): Promise<Employee | null> {
    return this.store.get(id) ?? null;
  }

  async save(employee: Employee): Promise<Employee> {
    if (employee.id === null) {
      employee.id = this.sequence++;
    }
    this.store.set(employee.id, employee);
    return employee;
  }

  async deleteById(id: number): Promise<void> {
    this.store.delete(id);
  }

  async existsById(id: number): Promise<boolean> {
    return this.store.has(id);
  }
}

